tests/test_executor.py

```
@@ -119,6 +121,114 @@ def test_executor_init_bad_tmp_path(
+    socket_dir = tmp_path_factory.getbasetemp() / "sock dir"
+    socket_dir.mkdir(exist_ok=True)
+
+    executor = PostgreSQLExecutor(
+        executable=pg_exe,
+        host=config.host,
+        port=port,
+        datadir=str(datadir),
+        unixsocketdir=str(socket_dir),
+        logfile=str(logfile_path),
+        startparams=config.startparams,
+        password="test pass",
```

Ruff (0.15.1)

[error] Possible hardcoded password assigned to argument: "password"

(S106)